applications/crossbar/src/modules_v1/cb_users_v1.erl

```diff
@@ -297,6 +341,180 @@ get_channels(Context) ->
+    case cb_context:resp_status(Attach) of
+        'error' -> JObj;
+        'success' ->
+            Data = cb_context:resp_data(Attach),
+            CT = wh_json:get_value(<<"content_type">>
+                                   ,wh_json:get_value(?PHOTO
+                                                      ,wh_json:get_value(<<"_attachments">>
+                                                                         ,cb_context:doc(Context)
+                                                                        ))),
+            wh_json:set_value(?PHOTO, wh_json:from_list([{CT, Data}]), JObj)
+    end.
+
+-spec set_org(wh_json:object(), cb_context:context()) -> wh_json:object().
+set_org(JObj, Context) ->
+    case wh_json:get_value(<<"org">>
+                           ,cb_context:doc(crossbar_doc:load(cb_context:account_id(Context)
+                                                             ,Context)))
+    of
+        'undefined' -> JObj;
+        Val -> wh_json:set_value(<<"org">>, Val, JObj)
+    end.
+
+-spec vcard_escape_chars(binary()) -> binary().
+vcard_escape_chars(Val) ->
+    Val1 = re:replace(Val, "(:|;|,)", "\\\\&", [global, {return, binary}]),
+    re:replace(Val1, "\n", "\\\\n", [global, {return, binary}]).
+
+vcard_fields_acc({_, Val}, Acc) when Val =:= 'undefined'; Val =:= []; Val =:= <<>> ->
+    Acc;
+vcard_fields_acc({Type, Val}, Acc) ->
+    case vcard_normalize_val(Val) of
+        <<>> -> Acc;
+        ValN ->
+            TypeN = vcard_normalize_type(Type),
+            [{TypeN, ValN} | Acc]
+    end;
+vcard_fields_acc([X | Rest], Acc) ->
+    vcard_fields_acc(Rest, vcard_fields_acc(X, Acc));
+vcard_fields_acc([], Acc) ->
+    Acc.
+
+-spec vcard_normalize_val(binary() | {char(), binaries()}) -> binary().
+vcard_normalize_val({Divider, Vals}) when is_list(Vals) ->
+    iolist_join(Divider, [vcard_escape_chars(X) || X <- Vals, X =/= 'undefined', X =/= [], X =/= <<>>]);
+vcard_normalize_val(Val) when is_binary(Val) ->
+    vcard_escape_chars(Val).
+
+-spec vcard_normalize_type(list() | {ne_binary(), ne_binary()} | ne_binary()) -> ne_binary().
+vcard_normalize_type(T) when is_list(T) -> iolist_join(<<";">>, [vcard_normalize_type(X) || X <- T]);
+vcard_normalize_type({T, V}) -> iolist_join(<<"=">>, [T, V]);
+vcard_normalize_type(T) -> T.
+
+%-spec card_field(ne_binary(), wh_json:object()) -> {vcard_type_spec(), vcard_val()}.
+card_field(Key, _)
+  when Key =:= <<"BEGIN">> ->
+    {Key, <<"VCARD">>};
+card_field(Key, _)
+  when Key =:= <<"VERSION">> ->
+    {Key, <<"3.0">>};
+card_field(Key, _)
+  when Key =:= <<"END">> ->
+    {Key, <<"VCARD">>};
+card_field(Key, JObj)
+  when Key =:= <<"FN">> ->
+    FirstName = wh_json:get_value(<<"first_name">>, JObj),
+    LastName = wh_json:get_value(<<"last_name">>, JObj),
+    MiddleName = wh_json:get_value(<<"middle_name">>, JObj),
+    {Key, iolist_join(<<" ">>, [X || X <- [FirstName, MiddleName, LastName], X =/= undefined, X =/= [], X =/= <<>>])};
```

Comment:
Maybe use `props:filter_empty([Firstname, MiddleName, Lastname])` instead of all the guards and the list comprehension?